CRM/Contact/BAO/Query.php

```diff
@@ -5054,7 +5054,12 @@ public function generatePermissionClause($onlyDeleted = FALSE, $count = FALSE) {
 
       if (CRM_Core_Permission::check('access deleted contacts')) {
         if (!$onlyDeleted) {
-          $this->_permissionWhereClause = str_replace('( 1 )', '(contact_a.is_deleted = 0)', $this->_permissionWhereClause);
+          if ($this->_permissionWhereClause === '( 1 )') {
+            $this->_permissionWhereClause = str_replace('( 1 )', '(contact_a.is_deleted = 0)', $this->_permissionWhereClause);
```

Comment:
This looks redundant to me.  There's no need to perform a str_replace when you know exactly what the string is.

I can't see from the logic above why this line shouldn't read

`$this->_permissionWhereClause = '(contact_a.is_deleted = 0)';`